sensor/openvas.go

```diff
@@ -0,0 +1,173 @@
+package sensor
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+)
+
+var processes = make(map[string]*os.Process)
+var mutex = &sync.Mutex{}
+
+// TODO: Remove global
+var sudo bool
+
+// StartScan starts scan with given scan-ID and process priority (-20 - 19,
+// lower is more prioritized)
+func StartScan(scan string, niceness int) error {
```

Comment:
```suggestion
func StartScan(scan string, niceness int) error {
```

Just pass the config Struct to Start Scan and extract the information (niceness, sudo, ...) from the config.

Reply:
I added the config in the scheduler as niceness is the only config StartScan needs. To determine the sudo option eulabeia tests if it can start openvas with sudo rights and determines if they are available. This is now done at the beginning of the scheduler function